Common/Data/Consolidators/WickedRenkoConsolidator.cs

```diff
@@ -0,0 +1,277 @@
+ * limitations under the License.
+*/
+
+using System;
+using QuantConnect.Data.Market;
+
+namespace QuantConnect.Data.Consolidators
+{
+    /// <summary>
+    /// This consolidator can transform a stream of <see cref="BaseData"/> instances into a stream of <see cref="RenkoBar"/>
+    /// with Renko type <see cref="RenkoType.Wicked"/>.
+    /// </summary>
+    public class WickedRenkoConsolidator : IDataConsolidator
+    {
+        private DateTime _closeOn;
+        private decimal _closeRate;
+        private bool _firstTick = true;
+        private decimal _highRate;
+        private RenkoBar _lastWicko;
+        private decimal _lowRate;
+        private DateTime _openOn;
+        private decimal _openRate;
+        private decimal _barSize;
+        private DataConsolidatedHandler _dataConsolidatedHandler;
+        private RenkoBar _currentBar;
+        private IBaseData _consolidated;
+
+        // Used for unit tests
+        internal RenkoBar OpenRenkoBar =>
+            new RenkoBar(null, _openOn, _closeOn, _barSize, _openRate, _highRate, _lowRate, _closeRate);
+
+        /// <summary>
+        /// Gets the kind of the bar
+        /// </summary>
+        public RenkoType Type { get; }
+
+        /// <summary>
+        /// Gets a clone of the data being currently consolidated
+        /// </summary>
+        public IBaseData WorkingData => _currentBar?.Clone();
+
+        /// <summary>
+        /// Gets the type consumed by this consolidator
+        /// </summary>
+        public Type InputType => typeof(IBaseData);
+
+        /// <summary>
+        /// Gets <see cref="RenkoBar"/> which is the type emitted in the <see cref="IDataConsolidator.DataConsolidated"/> event.
+        /// </summary>
+        public Type OutputType => typeof(RenkoBar);
+
+        /// <summary>
+        /// Gets the most recently consolidated piece of data. This will be null if this consolidator
+        /// has not produced any data yet.
+        /// </summary>
+        public IBaseData Consolidated
+        {
+            get { return _consolidated; }
+            private set { _consolidated = value; }
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="WickedRenkoConsolidator"/> class using the specified <paramref name="barSize"/>.
+        /// </summary>
+        /// <param name="barSize">The constant value size of each bar</param>
+        public WickedRenkoConsolidator(decimal barSize) // For use with WickedType
+        {
+            _barSize = barSize;
+            Type = RenkoType.Wicked;
+        }
+
+        /// <summary>
+        /// Event handler that fires when a new piece of data is produced
+        /// </summary>
+        public event EventHandler<RenkoBar> DataConsolidated;
+
+        /// <summary>
+        /// Event handler that fires when a new piece of data is produced
+        /// </summary>
+        event DataConsolidatedHandler IDataConsolidator.DataConsolidated
```

Comment:
nit - The same comment as the classic renko, let's move these last 2 guys above the constructor